### PR TITLE
Add go toolchain

### PR DESCRIPTION
**Pull request by pavel-the-best**

### CODE REVIEW

File: toolchains/go/detect.sh [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    [ "$1" = "$(printf '%s\n%s' "$1" "$2" | sort -V | head -n1)" ]
+}
+
+go_path="$( command -v go )"
+if [[ -z "$go_path" ]]; then
+    echo "go not found";
+    exit 1;
+fi
+version="$(go version | grep -Po "([0-9]\.?){3,}")"
+if [[ $(versionle "$version" "1.13") || "$version" == "1.13" ]]; then
```
Comment (MikailBag) [inline toolchains/go/detect.sh]: So, you reject $version == 1.13?

File: toolchains/go/detect.sh [hunk 2]
```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env bash
+
+versionle() {
+    [ "$1" = "$(printf '%s\n%s' "$1" "$2" | sort -V | head -n1)" ]
+}
+
+go_path="$( command -v go )"
```
Comment (MikailBag) [inline toolchains/go/detect.sh]: Add env var, e.g. `JJS_GO_SKIP` here. (as in other toolchains).

File: toolchains/go/detect.sh (hunk 1 above)
Comment (sleirsgoevy) [inline toolchains/go/detect.sh]: You probably meant ` && "$version" != "1.13"`

File: toolchains/go/detect.sh [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+if [[ -z "$go_path" ]]; then
+    echo "go not found";
+    exit 1;
+fi
+version="$(go version | grep -Po "([0-9]\.?){3,}")"
+if [[ $(versionle "$version" "1.13") || "$version" == "1.13" ]]; then
+    echo "need go at least version 1.13"
+    exit 1;
+fi
+go env -w GOPATH="$PWD"
```
Comment (MikailBag) [inline toolchains/go/detect.sh]: Are you sure $PWD is absolute path?

File: toolchains/go/detect.sh [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+go env -w GOROOT=
+$go_path get -u golang.org/x/tools/go/packages
```
Comment (sleirsgoevy) [inline toolchains/go/detect.sh]: escape `"$go_path"`

File: toolchains/go/detect.sh [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+$go_path run "$DATA/generate.go" > "program.go"
```
Comment (sleirsgoevy) [inline toolchains/go/detect.sh]: escape `"$go_path"`

File: toolchains/go/use/simple.sh [hunk 6]
```diff
@@ -0,0 +1,2 @@
+#!/usr/bin/env bash
+$GO run "program.go"
```
Comment (sleirsgoevy) [inline toolchains/go/use/simple.sh]: escape `"$GO"`

File: toolchains/go/detect.sh [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+if [[ -z "$go_path" ]]; then
+    echo "go not found";
+    exit 1;
+fi
+version="$(go version | grep -Po "([0-9]\.?){3,}")"
+if [[ $(versionle "$version" "1.13") && "$version" != "1.13" ]]; then
+    echo "need go at least version 1.13"
+    exit 1;
+fi
+go env -w GOPATH="$(env pwd)"
```
Comment (sleirsgoevy) [inline toolchains/go/detect.sh]: why not just `"$PWD"`

Comment (pavel-the-best) [reply toolchains/go/detect.sh]: Look at man pwd

**Comment by MikailBag** [comment]
Please re-send to jjs-dev/toolchains

**Pull request closed without merging**